webapp/functions.php

```diff
@@ -1,42 +1,5 @@
 <?php include "db.php";
 
-function showData($attribute) {
-    global $connection;
-    $query = "SELECT * FROM datapoints GROUP BY " . $attribute;
-    $result = mysqli_query($connection, $query);
-    if (!$result) {
-        die('Query FAILED' . mysqli_error());
-    }
-    
-    while($row = mysqli_fetch_assoc($result)) {
-        $data = $row[$attribute];
-                           
-        echo "<option value=''>$data</option>";
-    }
-}
-
-function showDayOfWeek($dateAttribute) {
-    global $connection;
-    $query = "SELECT * FROM datapoints GROUP BY " . $dateAttribute;
-    $result = mysqli_query($connection, $query);
-    if (!$result) {
-        die('Query FAILED' . mysqli_error());
-    }
-    
-    while($row = mysqli_fetch_assoc($result)) {
-        $date = $row[$dateAttribute];
-        
-        $unixTimestamp = strtotime($date);
-        
-        $weekday = date("l", $unixTimestamp);
-            
-                           
-        echo "<option value=''>$weekday</option>";
-    }
-}
-
-<?php include "db.php";
-

```

Comment:
Does this show all data? Good starting point but maybe abstract the query function as a separate function that this function calls. Then add a @filters array to the inputs of that query function. I think we have to define all the filters that we want to include in the search function and when a specific search configuration is submitted with some of those filters used (for example time and user id), we then go over each individual filter and if it exists add it to the query string. For example:
```
...
if (! is_null($filters[time][start])) {
  $query .= "WHERE starttime > " . $filters[time][start];
}
...
```
Or something like this.